rero_ils/modules/documents/serializers/ris.py

```diff
@@ -0,0 +1,171 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""RIS Document serialization."""
+
+from flask import current_app, request, stream_with_context
+from invenio_i18n.ext import current_i18n
+
+from rero_ils.modules.documents.api import Document
+from rero_ils.modules.documents.serializers.base import \
+    BaseDocumentFormatterMixin
+from rero_ils.modules.documents.utils import create_contributions
+from rero_ils.utils import get_i18n_supported_languages
+
+
+class RISSerializer:
+    """BibTeX serializer for records."""
+
+    def serialize(self, pid, record, links_factory=None, **kwargs):
+        """Serialize a single record and persistent identifier.
+
+        :param pid: Persistent identifier instance.
+        :param record: Record instance.
+        :param links_factory: Factory function for record links.
+        """
+        Document.post_process(record)
+        record = record.replace_refs()
+        contributions = create_contributions(
+            record.get('contribution', [])
+        )
+        if contributions:
+            record['contribution'] = contributions
+        return RISFormatter(record=record).format()
+
+    def serialize_search(self, pid_fetcher, search_result, links=None,
+                         item_links_factory=None, **kwargs):
+        """Serialize a search result.
+
+        :param pid_fetcher: Persistent identifier fetcher.
+        :param search_result: Elasticsearch search result.
+        :param links: Dictionary of links to add to response.
+        """
+        records = [
+            RISFormatter(record=hit['_source']).format()
+            for hit in search_result['hits']['hits']
+        ]
+
+        return stream_with_context(records)
+
+
+class RISFormatter(BaseDocumentFormatterMixin):
+    """RIS formatter class."""
+
+    def __init__(self, record, doctype_mapping=None, export_fields=None):
+        """Initialize RIS formatter with the specific record."""
+        super().__init__(record)
+        config = current_app.config.get('RERO_ILS_EXPORT_MAPPER').get('ris')
+        language = request.args.get("lang", current_i18n.language)
+        if not language or language not in get_i18n_supported_languages():
+            language = current_app.config.get('BABEL_DEFAULT_LANGUAGE', 'en')
+        self._language = language
+        self._doctype_mapping = doctype_mapping \
+            or config.get('doctype_mapping')
+        self._export_fields = export_fields or config.get('export_fields')
+
+    def format(self):
+        """Return RIS export for single record."""
+        return self._format()
+
+    def _format(self):
+        """Format record to RIS."""
```

Comment:
I think we can delete the function `def format(self):` and rename ` def _format(self):` directly to `def format(self):`?

Reply:
done